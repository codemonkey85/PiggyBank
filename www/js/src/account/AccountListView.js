"use strict"

import { AccountView } from "./AccountView.js"

export class AccountListView extends Backbone.View {
  constructor(args) {
    super(args)
  }

  preinitialize() {
    this.tagName = "ul"
    this.className = "accounts"
  }

  edit(model) {
    this.trigger("account:edit", model)
  }

  create(model) {
    this.trigger("account:create", model)
  }

  delete(model) {
    this.trigger("account:delete", model)
  }

  render() {
    for (const model of this.collection) {
      const view = new AccountView({ model })
      this.$el.append(view.render().el)
      this.listenTo(view, "account:edit", this.edit)
      this.listenTo(view, "account:create", this.create)
      this.listenTo(view, "account:delete", this.delete)
    }

    return this
  }
}
