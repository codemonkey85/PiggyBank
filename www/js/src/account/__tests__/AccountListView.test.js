import { AccountListView } from "../AccountListView"

test("AccountListView has render method", () => {
  const model = {
    get: jest.fn()
  }
  const collection = [model]
  const view = new AccountListView({ collection })
  const renderedView = view.render()
  expect(view).toBe(renderedView)
  expect(view.el).toMatchSnapshot()
})