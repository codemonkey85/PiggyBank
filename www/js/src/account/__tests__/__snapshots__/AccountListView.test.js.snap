// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountListView has render method 1`] = `
<ul
  class="accounts"
>
  <li
    class="account"
  >
    <!---->
    

    <div
      class="display"
    >
      
  
      <a
        href="/ledger/undefined"
      >
        <!---->
        <!---->
      </a>
      
  
      <button
        class="edit btn btn-sm btn-icon btn-outline-secondary"
      >
        <span
          class="icon-pencil"
        />
      </button>
      
  
      <button
        class="create btn btn-sm btn-icon btn-outline-primary"
      >
        <span
          class="icon-plus"
        />
      </button>
      
  
      <button
        class="delete btn btn-sm btn-icon btn-outline-danger"
      >
        <span
          class="icon-trash"
        />
      </button>
      

    </div>
    

    <!---->
  </li>
</ul>
`;
